src/puya/errors.py

```diff
@@ -34,16 +34,23 @@ class CodeError(PuyaError):
 def log_exceptions(fallback_location: SourceLocation | None = None) -> Iterator[None]:
     try:
         yield
-    except CodeError as ex:
-        logger.error(ex.msg, location=ex.location or fallback_location)  # noqa: TRY400
-    except InternalError as ex:
-        _log_traceback()
-        logger.critical(ex.msg, location=ex.location or fallback_location)
-        sys.exit(ErrorExitCode.internal)
-    except Exception as ex:
-        _log_traceback()
-        logger.critical(f"{type(ex).__name__}: {ex}", location=fallback_location)
-        sys.exit(ErrorExitCode.internal)
+    except* CodeError as code_errors:
+        for code_error in code_errors.exceptions:
+            assert isinstance(code_error, CodeError)
+            logger.error(code_error.msg, location=code_error.location or fallback_location)  # noqa: TRY400
+    except* InternalError as internal_errors:
+        for internal_error in internal_errors.exceptions:
+            assert isinstance(internal_error, InternalError)
+            _log_traceback()
```

Comment:
Given this doesn't take any parameters, would this make sense to call outside the loop instead?